Update the Security Policy on the WSO2 Ack Program (#3095)
ballerina-platform/ballerina-dev-website: security-policy.md

---
layout: ballerina-inner-page
title: Security Policy
intro: We take security issues very seriously and all the vulnerability reports are treated with the highest priority and confidentiality.
permalink: /security-policy/
redirect_from:
    - /security/
    - /security
    - /security-policy
---

Thank you for taking the time to [responsibly disclose](https://en.wikipedia.org/wiki/Responsible_disclosure) any vulnerabilities you find.

## Reporting a Vulnerability

> First of all, make sure that you are using the latest Ballerina version before you run an automated security scan or perform a penetration test against it. All security vulnerabilities and any other concerns related to security should be reported by sending an emailing to [security@ballerina.io](mailto:security@ballerina.io).  

Reviewer: ```suggestion
First of all, make sure that you are using the latest Ballerina version before you run an automated security scan or perform a penetration test against it. All security vulnerabilities and any other concerns related to security should be reported by sending an email to [security@ballerina.io](mailto:security@ballerina.io).  
```

> **WARNING:** To ensure end-user security, **please do not use any other medium to report security vulnerabilities**. Also, kindly refrain from disclosing the vulnerability details you come across with others, in any forums, sites, or other groups (public or private) before its mitigation and disclosure processes are completed.

If you would like, you can encrypt your report using the  following public key to send secure messages to [security@ballerina.io](mailto:security@ballerina.io):

> > security@ballerina.io: 0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 

Reviewer 2: ```suggestion
> security@ballerina.io: AC48 3C56 C0A0 6020 4BBE F3E4 182F 3F21 255F CCE9 
```

This key can also be found at [keys.openpgp.org](https://keys.openpgp.org/vks/v1/by-fingerprint/AC483C56C0A060204BBEF3E4182F3F21255FCCE9).

Also, use the following template when reporting vulnerabilities so that it contains all the required information that helps expedite the analysis and mitigation process.

- **Vulnerable Ballerina artifact(s) and version(s):** list  the vulnerable Ballerina artifact(s) and version(s) 
- **Overview:** provide a high-level overview of the issue and self-assessed severity
- **Description:** include the steps to reproduce
- **Impact:** state self-assessed impact
- **Solution:** propose a  solution if you have one

> We will keep you informed of the progress towards a fix and disclosure of the vulnerability if the reported issue is identified as a true positive.

Reviewer: ```suggestion
We will keep you informed of the progress towards fixing and disclosing of the vulnerability if the reported issue is identified as a true positive.
```

## Handling a Vulnerability

Here is an overview of our approach to handling vulnerabilities:

1. The vulnerability will be reported privately to our security team at [security@ballerina.io](mailto:security@ballerina.io).
2. Your email will be acknowledged within 24 hours, and you’ll receive a detailed response to your email indicating the next steps in handling your report. You will be updated on the progress. 

> 3. If case of a true-positive finding, the reported vulnerability will be confirmed and fixed by the relevant teams at WSO2.

Reviewer: ```suggestion
3. In case of a true-positive finding, the reported vulnerability will be confirmed and fixed by the relevant teams at WSO2.
```

> 4. The fix will be applied to the main branch and a new version of the distribution will be released immediately if required.

Author: ```suggestion
4. The fix will be applied to the main branch and a new release will be done immediately if required.
```

Reviewer: +1

> 4. The fix will be applied to the main branch and a new version of the distribution will be released immediately if required.

Reviewer: Is it correct to say new version of distribution here? As this is same regarding any component/website etc

Reviewer 2: Yes, it should be changed. How about something like this?
"4. The fix will be applied to the affected components and a new version will be released immediately if required."

Reviewer: +1

Reviewer 2: ```suggestion
4. The fix will be applied to the affected components and a new version will be released immediately if required.
```

5. The reported user is kept updated on the progress of the process. 

## Getting Acknowledged and Rewarded

Your efforts in reporting vulnerabilities or any other issues related to the security of Ballerina will be recognized and honored via the [WSO2 Security Reward and Acknowledgement Program](https://docs.wso2.com/display/Security/WSO2+Security+Reward+and+Acknowledgement+Program). 

> >**Note:** The reward program is currently applicable to vulnerabilities reported only in compiler, runtime, CLI tooling, standard library, VS Code plugin and [website](https://ballerina.io).

Reviewer: ```suggestion
>**Note:** The reward program is currently applicable to vulnerabilities reported only in the compiler, runtime, CLI tooling, standard library, VS Code plugin and,  [website](https://ballerina.io).
```

> security@ballerina.io: 0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 

This key can also be found at [keys.openpgp.org](https://keys.openpgp.org/vks/v1/by-fingerprint/AC483C56C0A060204BBEF3E4182F3F21255FCCE9).

Also, use the following template when reporting vulnerabilities so that it contains all the required information that helps expedite the analysis and mitigation process.

- **Vulnerable Ballerina artifact(s) and version(s):** list  the vulnerable Ballerina artifact(s) and version(s) 
- **Overview:** provide a high-level overview of the issue and self-assessed severity
- **Description:** include the steps to reproduce
- **Impact:** state self-assessed impact
- **Solution:** propose a  solution if you have one

We will keep you informed of the progress towards a fix and disclosure of the vulnerability if the reported issue is identified as a true positive.

## Handling a Vulnerability

Here is an overview of our approach to handling vulnerabilities:

1. The vulnerability will be reported privately to our security team at [security@ballerina.io](mailto:security@ballerina.io).

> 

Author: ```suggestion
```

2. Your email will be acknowledged within 24 hours, and you’ll receive a detailed response to your email indicating the next steps in handling your report. You will be updated on the progress. 
3. If case of a true-positive finding, the reported vulnerability will be confirmed and fixed by the relevant teams at WSO2.
4. The fix will be applied to the main branch and a new version of the distribution will be released immediately if required.

> 
> 

Author: ```suggestion
```

> 
> 
> 

Author: ```suggestion
```

permalink: /security-policy/
redirect_from:
    - /security/
    - /security
    - /security-policy
---

Thank you for taking the time to [responsibly disclose](https://en.wikipedia.org/wiki/Responsible_disclosure) any vulnerabilities you find.

## Reporting a Vulnerability

First of all, make sure that you are using the latest Ballerina version before you run an automated security scan or perform a penetration test against it. All security vulnerabilities and any other concerns related to security should be reported by sending an emailing to [security@ballerina.io](mailto:security@ballerina.io).  

> **WARNING:** To ensure end-user security, **please do not use any other medium to report security vulnerabilities**. Also, kindly refrain from disclosing the vulnerability details you come across with others, in any forums, sites, or other groups (public or private) before its mitigation and disclosure processes are completed.

If you would like, you can encrypt your report using the  following public key to send secure messages to [security@ballerina.io](mailto:security@ballerina.io):

> security@ballerina.io: 0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 

> This key can also be found at [pgp.mit.edu](https://pgp.surfnet.nl/pks/lookup?op=vindex&fingerprint=on&search=0xE683061E2F85C381).

Reviewer 2: ```suggestion
This key can also be found at [keys.openpgp.org](https://keys.openpgp.org/vks/v1/by-fingerprint/AC483C56C0A060204BBEF3E4182F3F21255FCCE9).
```

Also, use the following template when reporting vulnerabilities so that it contains all the required information that helps expedite the analysis and mitigation process.

- **Vulnerable Ballerina artifact(s) and version(s):** list  the vulnerable Ballerina artifact(s) and version(s) 
- **Overview:** provide a high-level overview of the issue and self-assessed severity
- **Description:** include the steps to reproduce
- **Impact:** state self-assessed impact
- **Solution:** propose a  solution if you have one

We will keep you informed of the progress towards a fix and disclosure of the vulnerability if the reported issue is identified as a true positive.

## Handling a Vulnerability

Here is an overview of our approach to handling vulnerabilities:

1. The vulnerability will be reported privately to our security team at [security@ballerina.io](mailto:security@ballerina.io).

> 

Reviewer: remove extra new lines

Author: Fixed via 99af5d5.

2. Your email will be acknowledged within 24 hours, and you’ll receive a detailed response to your email indicating the next steps in handling your report. You will be updated on the progress. 
3. If case of a true-positive finding, the reported vulnerability will be confirmed and fixed by the relevant teams at WSO2.
4. The fix will be applied to the main branch and a new version of the distribution will be released immediately if required.

> 

Reviewer: Extra new lines?

Author: Fixed via 99af5d5.

5. The reported user is kept updated on the progress of the process. 

## Getting Acknowledged and Rewarded

Your efforts in reporting vulnerabilities or any other issues related to the security of Ballerina will be recognized and honored via the [WSO2 Security Reward and Acknowledgement Program](https://docs.wso2.com/display/Security/WSO2+Security+Reward+and+Acknowledgement+Program). 

> >**Note:** The reward program is currently applicable to vulnerabilities reported only in the following applications and code repositories:
> 
> - [Ballerina website](https://ballerina.io/)
> - [Ballerina platform](https://github.com/ballerina-platform)

Reviewer 2: This seemed to be too generic and also feels like everything is covered with this clause. Shall we update as follows (assuming Ballerina central needs to be excluded as per the previous discussion):
```suggestion
>**Note:** The reward program is currently applicable to vulnerabilities reported only in compiler, runtime, CLI tooling, standard libraries, VS Code plugin and [website](https://ballerina.io).
```

Author: +1

Reviewer: +1

Reviewer: Should be `standard library` not libraries.

permalink: /security-policy/
redirect_from:
    - /security/
    - /security
    - /security-policy
---

Thank you for taking the time to [responsibly disclose](https://en.wikipedia.org/wiki/Responsible_disclosure) any vulnerabilities you find.

## Reporting a Vulnerability

First of all, make sure that you are using the latest Ballerina version before you run an automated security scan or perform a penetration test against it. All security vulnerabilities and any other concerns related to security should be reported by sending an emailing to [security@ballerina.io](mailto:security@ballerina.io).  

> **WARNING:** To ensure end-user security, **please do not use any other medium to report security vulnerabilities**. Also, kindly refrain from disclosing the vulnerability details you come across with others, in any forums, sites, or other groups (public or private) before its mitigation and disclosure processes are completed.

If you would like, you can encrypt your report using the  following public key to send secure messages to [security@ballerina.io](mailto:security@ballerina.io):

> security@ballerina.io: 0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 

> This key can also be found at [pgp.mit.edu](https://pgp.surfnet.nl/pks/lookup?op=vindex&fingerprint=on&search=0xE683061E2F85C381).

Reviewer 3: `pgp.surfnet.nl` isn't a well know. Let's please upload the public key also to `https://keys.openpgp.org/` and get the openpgp.org link listed here. 

In addition, shall we discuss with @anupama-pathirage if it's fine for the security team to manage the key in this instance (since security team is managing key for all other security reporting channels operated within WSO2). If agreeable, we'll re-issue a key-pair and share the new key with @anupama-pathirage (and take care of managing the key from that point onward).

Reviewer 2: +1 for that option. Also, the existing key is already expired and we may have to regenerate it again. In that case, it would be better to manage a new key-pair by security team going forward IMO.
@anupama-pathirage Shall we proceed with that?

Reviewer: Yes +1

Reviewer 2: @ayomawdb Let's proceed with this. We may have to update both ballerina.io/security and wso2.com/security pages along with this.

Reviewer 3: Ack.. I'll share a the public-key over here.

Reviewer 3: Links to new public key can be found in https://docs.wso2.com/display/Security/WSO2+Security+Vulnerability+Reporting+Guidelines

Reviewer 2: Thanks @ayomawdb for the prompt action.

Reviewer 2: @praneesha let's update this as 'This key can also be found at [keys.openpgp.org](https://keys.openpgp.org/vks/v1/by-fingerprint/AC483C56C0A060204BBEF3E4182F3F21255FCCE9).

Author: Fixed via https://github.com/ballerina-platform/ballerina-dev-website/pull/3095/commits/4ea2ceb06d5078e86614848ffef76c7731ebd8cb.

Also, use the following template when reporting vulnerabilities so that it contains all the required information that helps expedite the analysis and mitigation process.

- **Vulnerable Ballerina artifact(s) and version(s):** list  the vulnerable Ballerina artifact(s) and version(s) 
- **Overview:** provide a high-level overview of the issue and self-assessed severity
- **Description:** include the steps to reproduce
- **Impact:** state self-assessed impact
- **Solution:** propose a  solution if you have one

We will keep you informed of the progress towards a fix and disclosure of the vulnerability if the reported issue is identified as a true positive.

## Handling a Vulnerability

Here is an overview of our approach to handling vulnerabilities:

1. The vulnerability will be reported privately to our security team at [security@ballerina.io](mailto:security@ballerina.io).

2. Your email will be acknowledged within 24 hours, and you’ll receive a detailed response to your email indicating the next steps in handling your report. You will be updated on the progress. 

> 3. The reported vulnerability will be confirmed and fixed by the security team at WSO2.

Reviewer 3: ```suggestion
3. If case of a true-positive finding, the reported vulnerability will be confirmed and fixed by the relevant teams at WSO2.
```

4. The fix will be applied to the main branch and a new version of the distribution will be released immediately if required.

> 

Reviewer 3: ```suggestion
```

> 3. The reported vulnerability gets fixed and the solution gets verified by the relevant teams at WSO2.

Reviewer 3: ```suggestion
```

> 3. The fix gets applied to the main branch and a new version of the distribution gets released if required.

Reviewer 3: ```suggestion
```

> 4. The reported user is kept updated on the progress of the process. 

Reviewer 3: ```suggestion
5. The reported user is kept updated on the progress of the process. 
```
